napari/components/_layer_slicer.py

```diff
@@ -10,19 +11,31 @@
 _ViewerSliceRequest = dict  # [Layer, _LayerSliceRequest]
 _ViewerSliceResponse = dict  # [Layer, _LayerSliceResponse]
 
+LOGGER = logging.getLogger("napari.components._layer_slicer")
+
 
 class _LayerSlicer:
     def __init__(self):
         self.events = EmitterGroup(source=self, ready=Event)
         self._executor: Executor = ThreadPoolExecutor(max_workers=1)
-        self._task: Optional[Future[_ViewerSliceResponse]] = None
+        self._layers_to_task: dict[
+            tuple[Layer], Future[_ViewerSliceResponse]
+        ] = {}
 
     def slice_layers_async(
         self, layers: Iterable[Layer], dims: Dims
     ) -> Future[_ViewerSliceResponse]:
         """This should only be called from the main thread."""
-        if self._task is not None:
-            self._task.cancel()
+        # Cancel any tasks that are slicing a subset of the layers
+        # being sliced now. This allows us to slice arbitrary sets of
+        # layers with some sensible and not too complex cancellation
+        # policy.
+        layer_set = set(layers)
+        for task_layers, task in self._layers_to_task.items():
+            if set(task_layers).issubset(layer_set):
+                LOGGER.debug('Cancelling task for %s', task_layers)
+                task.cancel()
+
```

Comment:
I'm wondering if this is out of date with your current approach? I had to copy my old code onto your new branch and there might be a few things you changed that I didn't catch/remove.

Reply:
Looks right. Plus the diff here is outdated and I can't see it in the files changed tab. So I think this is just a temporary merge artifact and should be fine.